client/mysql.cc

```diff
@@ -3533,6 +3535,30 @@ print_field_types(MYSQL_RES *result)
 }
 
 
+static my_bool
+is_binary_field(MYSQL_FIELD *field)
+{
+  return ((field->type == MYSQL_TYPE_BLOB)
+    || ((field->type == MYSQL_TYPE_VAR_STRING) && (field->charsetnr == 63))
+    || ((field->type == MYSQL_TYPE_STRING) && (field->charsetnr == 63))
+    || ((field->type == MYSQL_TYPE_GEOMETRY) && (field->charsetnr == 63)));
```

Comment:
So it only applies to my_charset_bin string. Per my understanding any other character set (even latin1) can contain control characters that may trash terminal. Shouldn't they be considered too?